Snowflake.Data/Core/RestRequest.cs

```diff
@@ -131,21 +131,44 @@ HttpRequestMessage IRestRequest.ToRequestMessage(HttpMethod method)
 
             return message;
         }
-
     }
 
-    class AuthnRequest
+    class AuthenticatorRequest
     {
         [JsonProperty(PropertyName = "data")]
-        internal AuthnRequestData data { get; set; }
+        internal AuthenticatorRequestData Data { get; set; }
+
+        public override string ToString()
+        {
+            return String.Format("AuthenticatorRequest {{data: {0} }}", Data.ToString());
+        }
+    }
+
+    class AuthenticatorRequestData
+    {
+        [JsonProperty(PropertyName = "ACCOUNT_NAME", NullValueHandling = NullValueHandling.Ignore)]
+        internal String AccountName { get; set; }
+
+        [JsonProperty(PropertyName = "AUTHENTICATOR")]
+        internal String Authenticator { get; set; }
+
+        [JsonProperty(PropertyName = "BROWSER_MODE_REDIRECT_PORT", NullValueHandling = NullValueHandling.Ignore)]
```

Comment:
what is this parameter?

Reply:
This is the parameter being sent to Snowflake when the random port is decided locally, so that GS can generate the page to redirect user token sent to driver.